automation_tools/__init__.py

```diff
@@ -1655,16 +1655,168 @@ def setenforce(mode):
     run('setenforce {0}'.format(mode))
 
 
-def host_pings(host, timeout=15):
+# =============================================================================
+# Satellite and Capsule Upgrade
+# =============================================================================
+def copy_ssh_key(from_host, to_host):
+    """This will generate(if not already) ssh-key on from_host
+    and copy that ssh-key to to_host. Beware that both hosts
+    should have authorized key added for test running host.
+
+    :param from_host: A string. Hostname on which the key to be generated and
+        to be copied from.
+    :param to_host: A string. Hostname on to which the ssh-key will be copied.
+
+    """
+    ssh_out = execute(lambda: run('ls .ssh/'), host=from_host)
+    if 'id_rsa.pub' not in ssh_out[from_host]:
+        execute(
+            lambda: run('ssh-keygen -f .ssh/id_rsa -t rsa -N \'\''),
+            host=from_host)
+    local('scp root@{0}:.ssh/id_rsa.pub .'.format(from_host))
+    local('cat id_rsa.pub | ssh -o \'StrictHostKeyChecking no\' '
+          'root@{0} \'cat >> .ssh/authorized_keys\''.format(to_host))
+
+
+def capsule_certs_generate(path=None, capsule=None):
+    """Generates Certs on Satellite for given Capsule.
+    And also copies those certs into capsule.
+
+    :param capsule: A string. Capsule FQDN/hostname.
+    :param path: A string. Path on capsule where capsule cert to be copied.
+        If not provided then it will be copied in home directory of capsule.
+
+    """
+    if path is None:
+        path = '/home/'
+    # Generate Capsule Certificate
+    run('capsule-certs-generate --capsule-fqdn {0} '
+        '--certs-tar {0}-certs.tar'.format(capsule))
+    # Copy the Certificate in Capsule. Ssh key should be already added.
+    if capsule is not None:
+        run('scp {0}-certs.tar root@{0}:{1}'.format(capsule, path))
+
+
+def sync_capsule_tools_repos_to_upgrade(admin_password=None):
+    """This syncs capsule and sat-tools repos in Satellite server.
+    For upgrading Capsule and Client in feature.
+
+    :param admin_password: A string. Defaults to 'changeme'.
+        Foreman admin password for hammer commands.
+
+    Following environment variable affects this function.
+    CAPSULE_REPO
```

Comment:
Definition lists must, by definition, be separated from surrounding documentation.